src/go/configgenerator/filterconfig/filter_gen_jwt_authn.go

```diff
@@ -108,16 +108,18 @@ var jaFilterGenFunc = func(serviceInfo *ci.ServiceInfo) (*hcmpb.HttpFilter, []*c
 			PadForwardPayloadHeader: serviceInfo.Options.JwtPadForwardPayloadHeader,
 		}
 
-		if len(provider.GetAudiences()) != 0 {
-			for _, a := range strings.Split(provider.GetAudiences(), ",") {
-				jp.Audiences = append(jp.Audiences, strings.TrimSpace(a))
+		if !serviceInfo.Options.DisableJwtAudCheck {
```

Comment:
This logic seems incorrect.

My understanding is that the user wants this behavior:
1) If audience is empty in config and `--disable_jwt_aud_check=true`, then don't check audience for this selector
2) If audience is empty in config and `--disable_jwt_aud_check=false`, then check for `defaultAudience` for this selector
3) If audience is specified in config, always check for specified audience in selector, regardless of `--disable_jwt_aud_check` flag.

This CL breaks 3

Reply:
In my mind, when the flag `--disable_jwt_aud_check` is used,   we just don't check `aud` field in JWT regardless the what is in the service config, period.   This is brutal,  but very easy to understand.   

So I don't agree with your 3).

Reply:
Why though?

User wants a new flag added because they want to disable JWT aud check for some selectors be adding empty aud ("") to their config. But we currently default empty aud to `defaultAudience`

User didn't ask for a flag to completely disable audience check. User just wants to disable this default behavior when empty aud is in their config.

A better flag to solve this issue is `--disable_default_empty_jwt_aud` or similar

Reply:
Think about it this way: User has full control over the config.

If they want to completely disable JWT aud checking, they can set `--disable_default_empty_jwt_aud=true` and then modify their API config so every selector has empty ("") audience.

No need for a bigger scoped `--disable_jwt_aud_check` flag

Reply:
My concern is:  users will not be able to find `--disable_default_empty_jwt_aud` flag before we learn our such behaviors,they have to do a lot of testing it and debugging to learn the behaviors,  then start to use the flag to only disable it.  

It will be much easier for users to find `--disable_jwt_aud_check` flag, since it is obvious, and brutal.

The key question is: `--disable_jwt_aud_check` flag will not support partial aud checking,  some methods checking "aud" and some other methods not checking.   I think this should be fine, usually they want the same behaviors for all methods.

Reply:
This is a niche feature. We should've told customers that we don't implement non-security related features any more.

Seems either option is OK. Completely disable aud check could be fine given "aud" is optional; so there shouldn't be any security issues? But customers do mention "a solution could be to allow both an empty string or no aud claim at all if this was provided: x-google-audiences: "".

My suggestion would be check back with the customers and implement whichever is simple.

Reply:
OK,  let us just wrap it up and finish it since it is already started.  

I will go with @nareddyt approach,  add a new flag with small scope,  to only disable using service_name with empty audiences.  Will update the doc to make it easy to find the flag (which is my main concern of this approach).

Reply:
The new flag name is called as `--disable_jwt_service_name_audience`

Reply:
Done